velox/exec/Task.h

```diff
@@ -406,10 +406,9 @@ class Task : public std::enable_shared_from_this<Task> {
 
   ContinueFuture requestPauseLocked(bool pause);
 
-  void requestTerminate() {
-    std::lock_guard<std::mutex> l(mutex_);
-    terminateRequested_ = true;
-  }
+  // Requests activity of 'this' to stop. The returned future will be
+  // realized when the last thread stops running for 'this'.
+  ContinueFuture requestTerminate();
```

Comment:
Task::terminate takes terminateState, but this method doesn't. What state does it use?

Is this test-only method? If so, perhaps, rename it for clarity and to avoid accidental usage in production code: requestTerminateTestOnly.